extmod/modurandom.c

```diff
@@ -219,14 +219,15 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_2(mod_urandom_uniform_obj, mod_urandom_uniform);
 STATIC mp_obj_t mod_urandom___init__(void) {
     // This module may be imported by more than one name so need to ensure
     // that it's only ever seeded once.
-    static bool seeded = false;
-    if (!seeded) {
-        seeded = true;
+    if (!MP_STATE_VM(mod_urandom_seeded)) {
```

Comment:
Yes this is exactly the bug that this change is fixing. It seems like soft-reset _should_ cause a re-seed, so this is now the expected behavior.

Is there a comment somewhere else that says the wrong thing (w.r.t. soft reset) or are you saying I should just add a note here?

Reply:
The line above says `// that it's only ever seeded once.` It's not wrong, just ambiguous.

Reply:
Ah yep, thanks. Fixed!

Reply:
This change looks good, but shouldn't it be in its own commit? To me it doesn't look particularly related to 3cf95f6

Reply:
Yes it could go in a separate commit.

But I don't see how it will re-seed on soft reset, because soft reset doesn't clear out the root pointers (also it's an int so shouldn't really go in the root pointer section, that will slow down the GC by unnecessarily tracing it).

Reply:
@andrewleech It's because the logic that used to prevent duplicate initialization is no longer available. There used to be this kind of weird feature were specifically built-in modules that had an `__init__` get stored in `sys.modules` (but regular built-ins do not).

That said, modrandom was already working around another quirk of that old behavior (`urandom` vs `random`), so I figured that showing a "correct" way that a module-with-init can address this was related to the same change.

Reply:
I still think the state should not be stored in the root pointer section (see above), and so this change to `modurandom.c` could be entirely dropped.

Reply:
OK I've removed this.

Will re-visit in the future with an `MP_REGISTER_STATE()` macro.